pkg/app/piped/cloudprovider/lambda/client.go

```diff
@@ -109,17 +109,32 @@ func (c *client) IsFunctionExist(ctx context.Context, name string) (bool, error)
 
 func (c *client) CreateFunction(ctx context.Context, fm FunctionManifest) error {
 	input := &lambda.CreateFunctionInput{
-		Code: &types.FunctionCode{
-			ImageUri: aws.String(fm.Spec.ImageURI),
-		},
-		PackageType:  types.PackageType("Image"),
-		Role:         aws.String(fm.Spec.Role),
 		FunctionName: aws.String(fm.Spec.Name),
+		Role:         aws.String(fm.Spec.Role),
+		MemorySize:   aws.Int32(fm.Spec.Memory),
+		Timeout:      aws.Int32(fm.Spec.Timeout),
 		Tags:         fm.Spec.Tags,
 		Environment: &types.Environment{
 			Variables: fm.Spec.Environments,
 		},
 	}
+	// Container image packing.
+	if fm.Spec.ImageURI != "" {
+		input.PackageType = types.PackageTypeImage
+		input.Code = &types.FunctionCode{
+			ImageUri: aws.String(fm.Spec.ImageURI),
+		}
+	}
+	// Zip packing which stored in s3.
+	if fm.Spec.S3Bucket != "" {
```

Comment:
It seems to be assumed that only one of `ImageURI` or `S3Bucket` is specified.
I'm gonna say we need to add a check to `FunctionManifestSpec.validate()` as below:

```go
	if fmp.ImageURI != "" && fmp.S3Bucket != "" {
		return fmt.Errof("error")
	}
```

Reply:
Thanks for the comment, I understand your thought. I did the same way before but the AWS SDK itself validates this created `input` when we pass it to CreateFunction API and returns error which emphasizes what is missing in detail so I guess we don't need to do more validation stuff.

Reply:
On the second thought, I think it's good, lets me adopt that 🙏

Reply:
addressed by https://github.com/pipe-cd/pipe/pull/2567/commits/f8992ba25df3a72ebbac51b3a759e02b29c6e582 🙏